### PR TITLE
better TLD parsing and clean related classes

### PR DESCRIPTION
**Pull request by austin007008**
Replaced TLD parsing with dedicated code for each method to greatly reduce object creation.

Cleaned some related classes.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
-        ArrayList<Integer> nulls = TLD.instancesOf(0, columnFamily, 1);
-        final int start = nulls.get(0) + 1;
-        // uid is from the null byte to the end of the cf
-        ByteSequence uidByteSequence = columnFamily.subSequence(start, columnFamily.length());
-
-        UID uid = UID.parseBase(uidByteSequence.toString());
-        if (uid.getBaseUid().length() != uidByteSequence.length()) {
-            return new ArrayByteSequence(columnFamily.subSequence(0, start) + uid.getBaseUid());
+    protected ByteSequence parsePointer(ByteSequence cf) {
+        int index = 0;
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java]: The TLD utility already has a method for `parseRootPointerFromFI(ByteSequence bytes)`, I think we should add a new method `parseRootPointerFromEvent(ByteSequence bytes)` and call that here.

File: warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java [hunk 2]
```diff
@@ -497,7 +495,7 @@ public void functional_ContentExpansionTest() throws ParseException {
         ASTJexlScript newScript = FunctionIndexQueryExpansionVisitor.expandFunctions(config, helper, helper2, script);
 
         Map<String,ExpressionFilter> expressionFilters = getExpressionFilters(newScript, new AttributeFactory(mockTypeMetadata));
-        filter = new TLDEventDataFilter(newScript, Sets.newHashSet("FOO", "BAR"), expressionFilters, null, null, -1, 1);
+        filter = new TLDEventDataFilter(newScript, new HashSet<>(Arrays.asList("FOO", "BAR")), expressionFilters, null, null, -1, 1);
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java]: I would have left the `Sets.newHashSet()` calls

File: warehouse/query-core/src/main/java/datawave/query/tld/TLD.java [hunk 3]
```diff
@@ -23,112 +19,202 @@
  * </ul>
  */
 public class TLD {
+    private final static char NULL_BYTE = '\u0000';
+    private final static char DOT = '.';
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/tld/TLD.java]: Sonar lint is suggesting the following
```suggestion
    private static final char NULL_BYTE = '\u0000';
    private static final char DOT = '.';
```

File: warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java (hunk 2 above)
Comment (austin007008) [reply warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java]: guava says to use the regular java constructor and that "newHashSet()" will likely be deprecated in the future.
Will be using "Set.of" as a hashset is not needed here.

File: warehouse/query-core/src/main/java/datawave/query/tld/TLD.java (hunk 3 above)
Comment (austin007008) [reply warehouse/query-core/src/main/java/datawave/query/tld/TLD.java]: done

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java (hunk 1 above)
Comment (austin007008) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java]: done with the name "parseDatatypeAndRootUidFromEvent" as we also need the datatype, not just the root uid

**Review by apmoriarty — DISMISSED** [review]

**Review by apmoriarty — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java [hunk 4]
```diff
@@ -1,34 +1,25 @@
...
 public class TLDEventFieldAggregator extends EventFieldAggregator {
     public TLDEventFieldAggregator(String field, EventDataQueryFilter filter, int maxNextCount, TypeMetadata typeMetadata, String defaultTypeClass) {
         super(field, filter, maxNextCount, typeMetadata, defaultTypeClass);
     }
 
+    /**
+     * Event Data Key Structure (row, cf='datatype\0uid', cq='field\0value')
+     *
+     * @param cf
+     *            - a ByteSequence representing the Key's ColumnFamily
```
Comment (avgAGB) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java:19]: tickytack, but you could use a @return annotation here for the bytesequence line here

**Review by avgAGB — DISMISSED** [review]

**Review by avgAGB — APPROVED** [review]

**Review by apmoriarty — APPROVED** [review]

**Pull request merged**